src/game/scenes/action/SandwichGameScene.ts

```diff
@@ -61,57 +65,58 @@ export default class SandwichGameScene extends MinigameScene {
       'ground',
       '/static/assets/sprites/sandwich-game/sandwich_plan_ground.png'
     )
+    this.load.image(
+      'btn_left_on',
+      '/static/assets/sprites/sandwich-game/btn_left_on.png'
+    )
   }
 
   public create() {
     super.create()
     this.resetClassVariables()
     this.initBackground()
+    this.createControls()
   }
 
   public update(time: number, delta: number): void {
     this.skies!.forEach(sky => {
       sky!.x -= 0.15
     })
 
-    if (
-      this.skies![this.skies!.length - 1].x +
-        this.skies![this.skies!.length - 1].width / gameStore.ratioResolution <
-      this.game.config.width
-    ) {
-      this.skies![this.skies!.length + 1] = this.add
-        .sprite(
-          this.skies![0].width / gameStore.ratioResolution,
-          Number(this.game.config.height),
-          'sky'
-        )
-        .setOrigin(0, 1)
-        .setScale(1 / gameStore.ratioResolution)
-    }
-
-    this.lastGround = this.grounds![this.grounds!.length - 1]
-
-    // If the right of the last ground is in the viewport, we need to add an other ground to avoid blank ground
-    if (
-      this.lastGround.x + this.lastGround.width / gameStore.ratioResolution <
-      this.game.config.width
-    ) {
-      this.grounds![this.grounds!.length] = this.add
-        .sprite(
-          this.grounds![0].width / gameStore.ratioResolution,
-          Number(this.game.config.height),
-          'ground'
-        )
-        .setOrigin(0, 1)
-        .setScale(1 / gameStore.ratioResolution)
-    }
+    Array.from([
+      this.skies,
+      this.buildings,
+      this.landscapes,
+      this.streetLights,
+      this.grounds,
+    ]).forEach(element => {
+      // Loop through props sprites to determine if we have to clone a props to avoid blank
+      if (
+        element![element!.length - 1].x +
+          element![element!.length - 1].width / gameStore.ratioResolution <
+        this.game.config.width
+      ) {
+        if (element == this.skies) {
+          this.createCloneBackgroundElement(element!, 'sky')
+        } else if (element == this.buildings) {
+          this.createCloneBackgroundElement(element!, 'building')
+        } else if (element == this.landscapes) {
+          this.createCloneBackgroundElement(element!, 'landscape')
+        } else if (element == this.streetLights) {
+          this.createCloneBackgroundElement(element!, 'streetLights')
+        } else if (element == this.grounds) {
+          this.createCloneBackgroundElement(element!, 'ground')
+        }
+      }
+    })
   }
 
   protected initListeners(): void {
     super.initListeners()
     Emitter.on(GameEvents.SandwichPicked, () => {
-      this.player!.anims.play(this.playerWinTexture, true, 0)
-      this.onSuccess()
+      this.playWinAnimation()
```

Comment:
Là d'ailleurs faudrait pas aussi `await this.playWinAnimation()` ? A moins que tu veuilles direct supprimer le sandwitch sans attendre la fin de l'animation

Reply:
Ca me parait plus cohérent de l'enlever directement puisque dans l'animation, le perso tient le sandwich dans la main.

Reply:
Ah okok je savais pas, ouais effectivement go direct supprime le gameobject 👍